packages/core/src/config.ts

```diff
@@ -166,9 +166,27 @@ export async function createInternalRsbuildConfig(): Promise<RsbuildConfig> {
       },
     },
     output: {
-      // TODO: easy to development at the moment
       filenameHash: false,
-      minify: false,
+      minify: {
+        js: true,
+        css: false,
+        jsOptions: {
+          minimizerOptions: {
+            mangle: false,
+            minify: false,
+            compress: {
+              defaults: false,
+              unused: true,
+              dead_code: true,
+              toplevel: true,
+            },
+            format: {
+              // TODO: SWC will still remove part comments, like concatenation comments.
```

Comment:
concatenation comments is important, we must keep them🥹

Reply:
Remove the semicolon then the comments will be reserved, I'll try to fix in webpack and Rspack first. https://play.swc.rs/?version=1.7.18&code=H4sIAAAAAAAAA%2BOy1tdXcPb3c3YMcfUDYhcFX3%2BXUB9XKwU9%2FeKiZP1EvZJiruT8vOIShUQFWwX1jNScnHyF8vyinBR1ay4uAroz81JSK2Am5Oek6mXmpeVrJGpacwEAiY3Ax3gAAAA%3D&config=H4sIAAAAAAAAA32VS5LjMAiG9zlFSuvZ9mYOMLs5g0qxkKNuWbgETuLqyt0H%2BZHOdKPsbL4fIxDgz8PxaN6pM7%2BPn%2FIoL6MrBOXxLhaaM7ubWAzPI1BX4sjm106ZKgouESym%2B0oMu9IDVy%2Bgt01uEiLBLt9sQ8wxzM8BA5bB8ZNFbB0OA2Sm%2BkGXktnI%2FXEOEYwFiP53k0NMu99T0I0VvGrghJjA5VfIOrIxM%2FRQNFmHKbmRwF5cgw9S50ioBql0YvB2LDjqguwjR8xiV7AH522HHjQWC3QcL6A6SjxxzCRJtrCH09T3S4d853BxaXKsucJtuRw5sgLPGIltmNRirLBViZU2qhyDLcBT0WK%2BY8wtvw8AKUNyRNkNaqEWRZDmasOXrjEH6XqeFS4jomaaoZfS2hiDVt9aHygc1ZAF%2FNRBLXD3AjeKQdGDhRCka1R8jdyd1bB1W2DQiNy0U9NYgW1O5sbrgLzAfyRR1lttU8h6ObcpzcMJ1cna3YHP6F8I5DoY27jI3riNbT5lD9Ig4FXJRCqoc8uW0aZl7f7sEJkU%2BabtE56e9saPRTq43H%2BN%2F7rTD5vADOinBW5%2Fi3rHdSVn2TYZbmy%2BlPtafxzERPq7u0saHxmv2Rzu%2FwBFs49HgwYAAA%3D%3D

Reply:
it's back.

Reply:
the comment can be deleted?

Reply:
It's not possible to only preserve concatenate debug comment and remove users' comments.

Reply:
I mean the TODO comment in your code

Reply:
cleared, no more TODO left.